packages/react-html/src/server/components/Html.tsx

```diff
@@ -16,6 +16,10 @@ export interface Asset {
   integrity?: string;
 }
 
+export interface ScriptAsset extends Asset {
+  type?: 'module' | 'text/javascript';
```

Comment:
I was tempted to do this instead:
```
  module?: boolean;
```

The [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#attr-type) say that arbitrary values are accepted, but just result in an unprocessed script.  So not really in the spirit of a explicitly scripty properties.

Interested in your thoughts either way.

Reply:
I know this just matches up to the HTML spec where type is either a mime type or this magic 'module' string, but it feels like such an awkward API that I'd prefer almost anything else. My initial intuition was to do `type?: 'module' | 'script';`, which is kind of how I've seen these "execution modes" described in other tools like Babel.

As another piece to the puzzle, how do you think we will specify `nomodule` variants for a bundle? I know you are probably not worried about that for your specific case where you just care about a dev mode, but using this feature in a production setting would almost certainly require the use of the `nomodule` pattern for older browsers. Maybe that can help guide us to a nicer API here?

Reply:
🤔 That's an interesting wrinkle. I'm on board with no leaking spec details into this API, so how about:

```
type?: 'module' | 'nomodule' | 'script';

// type: 'module'
<script src="/foo.js" type="module" crossorigin="anonymous">

// type: 'nomodule'
<script src="/foo.js" type="text/javascript" nomodule crossorigin="anonymous">

// type: 'script' / default behaviour
<script src="./foo.js" type="text/javascript" crossorigin="anonymous">
```

Reply:
I like this API 👍